File: ANGEN/SampleFramework.Controls/SmoothControls - User's Guide/Samples/Snap UI Category/Fake IconEditor/CS/FakeIconEditor/BlendForm.cs
using Genetibase.Shared.Controls;

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace FakeIconEditor
{
	internal sealed partial class BlendForm : Form
	{
		public event EventHandler<ColorEventArgs> BlendColorChanged;

		private void OnBlendColorChanged(ColorEventArgs e)
		{
			if (this.BlendColorChanged != null)
			{
				this.BlendColorChanged(this, e);
			}
		}

		public void SetGradientBegin(Color gradientBegin)
		{
			_blendSelector.UpperColor = gradientBegin;
		}

		public void SetGradientEnd(Color gradientEnd)
		{
			_blendSelector.LowerColor = gradientEnd;
		}

		private void _blendSelector_ColorChanged(object sender, EventArgs e)
		{
			this.OnBlendColorChanged(new ColorEventArgs(_blendSelector.SelectedColor));
		}

		public BlendForm()
		{
			this.InitializeComponent();
		}
	}
}

File: ANGEN/SampleFramework.Controls/NavigationBar/Samples/Snap UI Category/Fake IconEditor/VB/FakeIconEditor/BlendForm.vb
Friend NotInheritable Class BlendForm
    Public Event BlendColorChanged As EventHandler(Of ColorEventArgs)

    Public Sub SetGradientBegin(ByVal gradientBegin As Color)
        _blendSelector.UpperColor = gradientBegin
    End Sub

    Public Sub SetGradientEnd(ByVal gradientEnd As Color)
        _blendSelector.LowerColor = gradientEnd
    End Sub

    Private Sub _blendSelector_ColorChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles _blendSelector.ColorChanged, _blendSelector.ColorChanging
        RaiseEvent BlendColorChanged(Me, New ColorEventArgs(Me._blendSelector.SelectedColor))
    End Sub
End Class